ebms-provider/src/main/kotlin/no/nav/emottak/ebms/configuration/Configurator.kt

```diff
@@ -0,0 +1,15 @@
+package no.nav.emottak.ebms.configuration
+
+import com.sksamuel.hoplite.ConfigLoader
+import com.sksamuel.hoplite.ExperimentalHoplite
+import com.sksamuel.hoplite.addEnvironmentSource
+import com.sksamuel.hoplite.addResourceSource
+
+@OptIn(ExperimentalHoplite::class)
+fun config() = ConfigLoader.builder()
+    .addEnvironmentSource()
+    .addResourceSource("/application-personal.conf", optional = true)
+    .addResourceSource("/application.conf")
+    .withExplicitSealedTypes()
+    .build()
+    .loadConfigOrThrow<Config>()
```

Comment:
Noen spørsmål så jeg forstår litt bedre:

Burde vi legge til `application-personal.conf` til gitignore så vi ikke committer det ved uhell?

Hvis jeg legger inn `/application-personal.conf`, blir den å overstyre `application.conf` for lokal testing? Har rekkefølgen vi setter disse noe å si? 

Hvor er det `keystoreLocation = "${KAFKA_KEYSTORE_PATH:-}"` blir satt? Altså, hvordan finner den verdien fra en variabel her? Tipper det er fra Vault, men sliter å se hvor denne konvensjonen blir gjort

Reply:
God ide å legge den til i gitignore. Så vidt jeg forstår får config som kommer først presedens. Står litt om det [her](https://github.com/sksamuel/hoplite#cascading-config).

Når [kafka.pool](https://doc.nais.io/workloads/application/reference/application-spec/#kafkapool) er lagt til i nais.yaml, blir environment variablene nevnt [her](https://doc.nais.io/persistence/kafka/reference/environment-variables/index.html) automatisk gjort tilgjengelig i podden av NAIS-plattformen. Vault er ikke involvert med kafka.

Reply:
Ah, så det er fra **env** de i config kommer fra altså!  Takk som forklarer!